### DISCUSSION TITLE
Help with shaders

### DISCUSSION THREAD

**Discussion by SammuXDD** [CodenameCrew/CodenameEngine#506] [Q&A] [answered]
So i'm trying to recreate Darnel bf mix and when i import the rain shader and code it to the game, but when i play the song the screen looks like dark grey.

![2024-12-16_17-06](https://github.com/user-attachments/assets/99151386-24ce-482d-bd52-9e3961ea29d1)

here is the code that i wrote (i put this code in a stage with a script:

var rainShader:CustomShader = null;

rainShader = new CustomShader('rain');
rainShader.uTime = 0;
rainShader.uScale = FlxG.height / 200;
rainShader.uIntensity = 0.2;
rainShader.uScreenResolutionV = [FlxG.width ,FlxG.height];

var obj = new FlxSprite(0, 0).makeGraphic(FlxG.width ,FlxG.height, FlxColor.WHITE);
obj.shader = rainShader;

FlxG.camera.setFilters([new ShaderFilter(obj.shader)]);

**Comment by NeeEoo** [comment, marked as answer]
Theres a port of it in the vslice-modpack pinned on the CodenameCrew organization page

**Reply by SammuXDD** [reply]
ok, i will take a look at it

**Reply by SammuXDD** [reply]
So i try that but it doesn't appear and if i add it to the stage it just gives me a blackScreen

**Reply by SammuXDD** [reply]
i modify the script from the repo and it now works , thank you so much :D
